Lib/test/test_uuid.py

```diff
@@ -34,6 +34,26 @@ def get_command_stdout(command, args):
 class BaseTestUUID:
     uuid = None
 
+    def test_nil_uuid(self):
+        nil_uuid = self.uuid.NIL
+
+        s = '00000000-0000-0000-0000-000000000000'
+        i = 0x00000000000000000000000000000000
+        self.assertEqual(nil_uuid, self.uuid.UUID(s))
+        self.assertEqual(nil_uuid, self.uuid.UUID(int=i))
+        self.assertEqual(nil_uuid.int, i)
+        self.assertEqual(str(nil_uuid), s)
+
+    def test_max_uuid(self):
+        max_uuid = self.uuid.MAX
+
+        s = 'ffffffff-ffff-ffff-ffff-ffffffffffff'
+        i = 0xffffffffffffffffffffffffffffffff
+        self.assertEqual(max_uuid, self.uuid.UUID(s))
+        self.assertEqual(max_uuid, self.uuid.UUID(int=i))
+        self.assertEqual(max_uuid.int, i)
+        self.assertEqual(str(max_uuid), s)
```

Comment:
```suggestion
        self.assertEqual(str(max_uuid), s)
        self.assertIsNone(max_uuid.version)
        self.assertEqual(max_uuid.variant, self.uuid.RESERVED_FUTURE)
```

Be careful: for NIL, the variant would be deduced to be `self.uuid.RESERVED_NCS` because everything is 0. However, I don't know whether we should return 0 for the version or None. What does the RFC say?

Reply:
The RFC explicitly states that the Nil UUID falls under the Apollo NCS variant and the Max UUID under the "yet-to-be defined" future UUID variant:

- https://www.rfc-editor.org/rfc/rfc9562.html#section-5.9-4
- https://www.rfc-editor.org/rfc/rfc9562.html#section-5.10-4
- https://www.rfc-editor.org/rfc/rfc9562.html#table1

Reply:
As for the version, all zeros is "Unused" and all ones is "Reserved for future definition", so returning `None` here for both Nil and Max is appropriate:

- https://www.rfc-editor.org/rfc/rfc9562.html#table2

Reply:
Can you add a comment with those links in the tests please?

Reply:
Sure. Incoming...